twisted/mail/test/test_mailmail.py

```diff
@@ -73,3 +100,129 @@ def test_copyAliasedSender(self):
         """
         exc = self.assertRaises(SystemExit, parseOptions, ['-om'])
         self.assertEqual(exc.args, ("Unsupported option.",))
+
+
+    def test_Version(self):
+        """
+        The I{--version} option displays the version and raises
+        L{SystemExit}.
+        """
+        out = StringIO()
+        self.patch(sys, 'stdout', out)
+        self.assertRaises(SystemExit, parseOptions, '--version')
+        data = out.getvalue()
+        self.assertEqual(data, "mailmail version: {}\n".format(version))
+
+
+    def test_backGroundDelivery(self):
+        """
+        The I{-odb} flag specifies background delivery.
+        """
+        self.addCleanup(setattr, sys, 'stdin', sys.stdin)
+        sys.stdin = StringIO('\n')
+        o = parseOptions("-odb")
+        self.assertTrue(o.background)
+
+
+    def test_foreGroundDelivery(self):
+        """
+        The I{-odf} flags specifies foreground delivery.
+        """
+        self.addCleanup(setattr, sys, 'stdin', sys.stdin)
+        sys.stdin = StringIO('\n')
+        o = parseOptions("-odf")
+        self.assertFalse(o.background)
+
+
+    def test_recipientsFromHeaders(self):
+        """
+        The I{-t} flags specifies that recipients should be obtained
+        from headers.
+        """
+        self.addCleanup(setattr, sys, 'stdin', sys.stdin)
+        sys.stdin = StringIO(
+            'To: Curly <invaliduser2@example.com>\n'
+            'Cc: Larry <invaliduser1@example.com>\n'
+            'Bcc: Moe <invaliduser3@example.com>\n'
+            '\n'
+            'Oh, a wise guy?\n')
+        o = parseOptions("-t")
+        self.assertEqual(len(o.to), 3)
+
+
+    def test_setFrom(self):
+        """
+        The I{-F} flags specifies the From: value.
+        """
+        self.patch(sys, 'stderr', self.out)
+        self.addCleanup(setattr, sys, 'stdin', sys.stdin)
+        sys.stdin = StringIO(
+            'To: invaliduser2@example.com\n'
+            'Subject: A wise guy?\n\n')
+        o = parseOptions(["-F", "Larry <invaliduser1@example.com>", "-t"])
+        self.assertEqual(o.sender, "Larry <invaliduser1@example.com>")
+        # Test that -F flag is overridden by From: value in header
+        sys.stdin = StringIO(
+            'To: Curly <invaliduser4@example.com>\n'
+            'From: Shemp <invaliduser4@example.com>\n')
+        o = parseOptions(["-F", "Groucho <invaliduser5@example.com>", "-t"])
+        self.assertEqual(o.sender, "invaliduser4@example.com")
+
+
+    def test_run(self):
+        """
+        Test run() method.
+        """
+        if platformType == "win32":
+            raise SkipTest("win32 lacks support for getuid()")
```

Comment:
this is strange. it looks like platformType is never 'win32' as this line is never executed in the test.

I think that in Twisted, tests should be skipped using `test_run.skip = "win32 lacks support for getuid()"`